addons/account/wizard/account_payment_register.py

```diff
@@ -414,42 +414,47 @@ def _compute_show_require_partner_bank(self):
+            'amount_residual': sentinel_balance,
+            'amount_residual_currency': sentinel_amount_currency,
+            'company': self.company_id,
+            'currency': self.currency_id,
+            'date': self.payment_date,
+        }
+        counterpart_vals_list = [
+            {
+                'balance': x.balance,
+                'amount_currency': x.amount_currency,
+                'amount_residual': x.amount_residual,
+                'amount_residual_currency': x.amount_residual_currency,
+                'company': x.company_id,
+                'currency': x.currency_id,
+                'date': x.date,
+            }
+            for x in batch_result['lines']
+        ]
+        self.env['account.move.line']._prepare_reconciliation_partials(counterpart_vals_list + [payment_vals])
+        return abs(sentinel_amount_currency - payment_vals['amount_residual_currency'])
+
     @api.depends('can_edit_wizard', 'source_amount', 'source_amount_currency', 'source_currency_id', 'company_id', 'currency_id', 'payment_date')
     def _compute_amount(self):
         for wizard in self:
             if wizard.source_currency_id and wizard.can_edit_wizard:
-                comp_curr = wizard.company_id.currency_id
-                if wizard.source_currency_id == wizard.currency_id:
-                    # Same currency.
-                    wizard.amount = wizard.source_amount_currency
-                elif wizard.source_currency_id != comp_curr and wizard.currency_id == comp_curr:
-                    # Foreign currency on source line but the company currency one on the opposite line.
-                    wizard.amount = wizard.source_currency_id._convert(
-                        wizard.source_amount_currency,
-                        comp_curr,
-                        wizard.company_id,
-                        wizard.payment_date,
-                    )
-                elif wizard.source_currency_id == comp_curr and wizard.currency_id != comp_curr:
-                    # Company currency on source line but a foreign currency one on the opposite line.
-                    batch = wizard._get_batches()[0]
-                    wizard.amount = abs(sum(
-                        comp_curr._convert(
-                            aml.amount_residual,
-                            wizard.currency_id,
-                            wizard.company_id,
-                            aml.date,
-                        )
-                        for aml in batch['lines']
-                    ))
-                else:
-                    # Foreign currency on payment different than the one set on the journal entries.
-                    wizard.amount = comp_curr._convert(
-                        wizard.source_amount,
-                        wizard.currency_id,
-                        wizard.company_id,
-                        wizard.payment_date,
-                    )
+                batch_result = wizard._get_batches()[0]
```

Comment:
Lines are batched. If you are paying 5 invoices having the same partner/currency, all will be part of the same batch. So here, since `can_edit_wizard` is True, it means there is only one batch to pay that could be multiple invoices

Reply:
Having the same partner, I'm ok (but not 100%, there is still the possibility to have different partner but same commercial_partner).
But the same currency, I'm not so sure.
For one partner, you could have several invoice/bill in different currencies.

At the moment, it's possible to select several one and to make one payment for all of them